File: Data_Structures/javascript/Stack/twoStacksInArray.js
class twoStack {
    constructor() {
        this.size = n;
        this.top1 = -1;
        this.top2 = n;
        this.arr = [];
    }

    // Push in the first Stack
    push1 = elm => {
        if (this.top1 < this.top2 - 1)
            this.arr[++this.top1] = elm;
        else
            return 'Stack Overflow';
    }

    push2 = elm => {
        if (this.top1 < this.top2 - 1)
            this.arr[--this.top2] = elm;
        else
            return 'Stack Overflow';
    }

    pop1 = () => {
        if (this.top1 >= 0) {
            let elm = this.arr[this.top1];
            this.top1--;
            return elm;
        } else {
            return 'Stack Underflow';
        }
    }

    pop2 = () => {
        if (this.top2 < this.size){
            let elm = this.arr[this.top2];
            this.top2++;
            return elm;
        } else {
          return 'stack underflow';
        }
    }
}

File: Data_Structures/C/Stack/twoStacksInArray.c
#include <stdlib.h>

struct twoStacks {
    int *arr;
    int size;
    int top1 = size / 2 + 1;
    int top2 = size / 2;
};

void push1(int x)
{
    if (top1 > 0) {
        top1--;
        arr[top1] = x;
    }
    else {
        cout << "Stack Overflow";
        return;
    }
}

void push2(int x)
{
    if (top2 < size - 1) {
        top2++;
        arr[top2] = x;
    }
    else {
        cout << "Stack Overflow";
        return;
    }
}

int pop1()
{
    if (top1 <= size / 2) {
        int x = arr[top1];
        top1++;
        return x;
    }
    else {
        cout << "Stack UnderFlow";
        exit(1);
    }
}

int pop2()
{
    if (top2 >= size / 2 + 1) {
        int x = arr[top2];
        top2--;
        return x;
    }
    else {
        cout << "Stack UnderFlow";
        exit(1);
    }
}
